### ISSUE TITLE
how to change default IP address and port number?

### ISSUE THREAD

**Issue by ulinktech** [jeelabs/esp-link#322]
hi 
i wanna change the default ip address and port. Now ip address is 192.168.4.1:23.
but can not change at http server. 
How to change it? 
My means that i want change IP address and port of esp8266 module.
pls inform to me.

regards

**Comment by uzi18** [comment]
Why can't You change it by http, it doesn't work?

2017-08-03 10:03 GMT+02:00 ulinktech <notifications@github.com>:

**Comment by ulinktech** [comment]
yes. we did changed it by http of static ip address then saved  but can't change it.
also cannot found change function of port number.

**Comment by uzi18** [comment]
Just reflash default options.

04.08.2017 12:49 AM "ulinktech" <notifications@github.com> napisał(a):

yes. we did changed it by http of static ip address then saved but can't
change it.
also cannot found change function of port number.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/jeelabs/esp-link/issues/322#issuecomment-320110590>,
or mute
the thread

**Comment by tve** [comment]
Changing the IP address of the AP is not supported. Most likely there is an SDK call you can insert somewhere to change it, but esp-link doesn't currently do that.

**Comment by kriegste** [comment]
Why would anyone want to change the AP IP anyway?

**Comment by ulinktech** [comment]
Our special apps are requires special IP and PORT. So, we have to change ip&port on esp8266 module.
But official  firmware was support ip&port change function. but official firmware was not support transparent function at server mode. how to ?

**Comment by kriegste** [comment]
There is an example in the SDK.

```
wifi_set_opmode(STATIONAP_MODE);

struct ip_info info;

wifi_softap_dhcps_stop();
 
IP4_ADDR(&info.ip, 10, 10, 10, 1); 
IP4_ADDR(&info.gw, 10, 10, 10, 1); 
IP4_ADDR(&info.netmask,  255, 255, 255, 0); 
wifi_set_ip_info(SOFTAP_IF, &info);

wifi_softap_dhcps_start();
```

You will have to implement that by yourself.

The problem with the AP IP is that if you lose it, there is no fallback. You can just "guess" what it was. The better solution would be access by hostname (even in AP mode) but I have no clue how to do that...

**Comment by ulinktech** [comment]
thanks for your reply. 
Let me take your suggestion . 
if any idea, pls inform to me.
